enclave-runner/src/usercalls/mod.rs

```diff
@@ -165,18 +165,23 @@ pub trait SyncStream: 'static + Send + Sync {
 }
 
 
-trait SyncListener: 'static + Send + Sync {
-    fn accept(&self) -> IoResult<(FileDesc, Box<dyn ToString>, Box<dyn ToString>)>;
+pub trait SyncListener: 'static + Send + Sync {
+    fn accept(&self, local_addr: Option<&mut String>, peer_addr: Option<&mut String>) -> IoResult<Box<dyn SyncStream>>;
 }
 
 impl SyncListener for TcpListener {
-    fn accept(&self) -> IoResult<(FileDesc, Box<dyn ToString>, Box<dyn ToString>)> {
+    fn accept(&self, local_addr: Option<&mut String>, peer_addr: Option<&mut String>) -> IoResult<Box<dyn SyncStream>> {
         TcpListener::accept(self).map(|(s, peer)| {
-            let local = match s.local_addr() {
-                Ok(local) => Box::new(local) as _,
-                Err(_) => Box::new("error") as _,
-            };
-            (FileDesc::stream(s), local, Box::new(peer) as _)
+            if let Some(local_addr) = local_addr {
+                match self.local_addr() {
+                    Ok(local) => local_addr.push_str(&local.to_string()),
```

Comment:
I think the intent is clearer when you write `*local_addr = local.to_string()`.

Do you think we should guarantee an empty string is passed in?

Reply:
I have made it cleaner in the current push by using `*local_addr = local.to_string()`
I guess, since the reference to the string originates in this file itself, we need not add an extra check to ensure it is empty.

Reply:
> since the reference to the string originates in this file itself, we need not add an extra check to ensure it is empty.

What do you mean? We call `accept` on user implementations of `SyncListener`. If we guarantee that an empty string is passed in, then we should put that in the function documentation.

Reply:
Got it. My earlier understanding of 
`we should guarantee an empty string is passed in` was incorrect ( i thought this function should error out if empty string is not passed in).
I will update the documentation.

Reply:
I have added the documentation changes in recent push.